### PR TITLE
📣 ❌  delete broadcasts mulligan

### PR DESCRIPTION
**Pull request by susanm74**

### CODE REVIEW

File: temba/msgs/tests.py [hunk 1]
```diff
@@ -2306,6 +2306,44 @@ def test_scheduled_update(self):
...
+            schedule=schedule,
+        )
+        broadcast = Broadcast.objects.get()
+        # print(f"broadcast={broadcast}")
+
+        delete_url = reverse("msgs.broadcast_scheduled_delete", args=[broadcast.id])
+        # print(f"delete_url={delete_url}")
+
+        # fetch the delete modal
+        response = self.client.get(delete_url)
```
Comment (norkans7) [inline temba/msgs/tests.py:2325]: Remove the debug line that are commented out

File: temba/msgs/tests.py [hunk 2]
```diff
@@ -2306,6 +2306,44 @@ def test_scheduled_update(self):
...
+        # self.assertContains(response, "You are about to delete")
+
+        # submit the delete modal
+        response = self.assertDeleteSubmit(delete_url, object_deleted=broadcast, success_status=200)
+        # print(f"delete_submit_response={response}")
+        # print(f"delete_submit_response_value={response.getvalue()}")
+        # print(f"delete_submit_response_items={response.items()}")
+        # TODO figure out why this is expecting "broadcast" and not "messages"
+        # or TODO figure out where "broadcast" is getting changed to "messages"
+        self.assertEqual("/broadcast/scheduled/", response["Temba-Success"])
```
Comment (norkans7) [inline temba/msgs/tests.py:2345]: We also need to assert that no broadcast with the `broadcast.id` still exist and same for the schedule with `schedule.id`

**Review by norkans7 — CHANGES_REQUESTED** [review]
a couple of changes for the tests

**Comment by susanm74** [comment]
Screen Shot 2022-10-11 at 12 33 06 PM

**Comment by susanm74** [comment]
Screen Shot 2022-10-11 at 12 33 27 PM

**Comment by susanm74** [comment]
how do we feel about the way the Broadcast object displays in the modal? let me know if we should customize this to be a bit cleaner for users (ex. i think users refer to these entities as "scheduled messages" and not "broadcasts", yeah? or do they refer to them using either name?)

**Comment by susanm74** [comment]
also, if the scheduled message has been sent at least once before being deleted, does deleting risk creating an orphan if there's a FK referring to a Broadcast that doesn't exist anymore? (ex. Msg and BroadcastMsgCount) i noticed these FKs are defined to PROTECT on_delete, but when i schedule a message, send it once, and then delete the scheduled message, i noticed that it lets me delete it, and it also deletes the sent message, and decrements the sent messages total - is this how we expect deleting a scheduled message should behave? or should this scenario disallow a scheduled message from being deleted?

**Comment by susanm74** [comment]
OLD INTERFACE
Screen Shot 2022-10-11 at 4 18 06 PM
Screen Shot 2022-10-11 at 4 18 17 PM

**Comment by susanm74** [comment]
NEW INTERFACE
Screen Shot 2022-10-11 at 4 18 42 PM

**Comment by susanm74** [comment]
> also, if the scheduled message has been sent at least once before being deleted, does deleting risk creating an orphan if there's a FK referring to a Broadcast that doesn't exist anymore? (ex. Msg and BroadcastMsgCount) i noticed these FKs are defined to PROTECT on_delete, but when i scheduled a message, sent it once, and then deleted the scheduled message, i noticed that it lets me delete it, and it also deletes the sent message, and decrements the sent messages total - is this how we expect deleting a scheduled message should behave? or should this scenario disallow a scheduled message from being deleted?

was going to shift this out of draft mode, but i want to be sure this scenario is being handled correctly...
@norkans7 pointed out that the release method is kind of bypassing the FK stuff and deleting all of the children, msgs, broadcastMsgCount, schedules, etc are being deleted before deleting the broadcast itself - is this the behavior we want? i think i see 3 options:
1. use the existing logic the way it is, although i could see this being confusing to a user to have sent messages suddenly disappear after deleting a scheduled message, but i'm not sure if there's an easy way to keep them around without keeping the broadcast around
2. make it super duper clear using language in the delete modal and specify what else will get deleted when a broadcast is deleted
3. update the logic to somehow archive or deactivate or hide the broadcast from the list of scheduled messages, but don't actually delete it

**Comment by susanm74** [comment]
@ericnewcomer @rowanseymour a gentle nudge here...

**Comment by susanm74** [comment]
closing in favor of:
- PR#1 - https://github.com/nyaruka/rapidpro/pull/4056
- PR#2 - TBD
- PR#3 - https://github.com/nyaruka/rapidpro/pull/4059
- PR#4 - https://github.com/nyaruka/rapidpro/pull/4057
- PR#5 - https://github.com/nyaruka/rapidpro/pull/4058

**Pull request closed without merging**